Microsoft.Toolkit.Uwp.UI/Cache/CacheBase.cs

```diff
@@ -0,0 +1,380 @@
+
+                    var msi = new InMemoryStorageItem<T>(fileName, properties.DateModified.DateTime, t);
+                    _inMemoryFileStorage.SetItem(msi);
+                }
+            }
+
+            return t;
+        }
+
+        private async Task<T> DownloadFileAsync(Uri uri, StorageFile baseFile, bool preCacheOnly)
+        {
+            T t = default(T);
+
+            using (var webStream = await StreamHelper.GetHttpStreamAsync(uri))
+            {
+                // if its pre-cache we aren't looking to load items in memory
+                if (!preCacheOnly)
+                {
+                    t = await InitializeTypeAsync(webStream);
+
+                    webStream.Seek(0);
+                }
+
+                using (var reader = new DataReader(webStream))
+                {
+                    await reader.LoadAsync((uint)webStream.Size).AsTask().ConfigureAwait(false);
+                    var buffer = new byte[(int)webStream.Size];
+                    reader.ReadBytes(buffer);
+                    await FileIO.WriteBytesAsync(baseFile, buffer).AsTask().ConfigureAwait(false);
+                }
+            }
+
+            return t;
+        }
+
+        private async Task<bool> IsFileOutOfDate(StorageFile file, DateTime expirationDate)
+        {
+            if (file == null)
+            {
+                return true;
+            }
+
+            var properties = await file.GetBasicPropertiesAsync().AsTask().ConfigureAwait(false);
+            return properties.DateModified < expirationDate;
+        }
+
+        private async Task InternalClearAsync(IEnumerable<StorageFile> files)
+        {
+            foreach (var file in files)
+            {
+                try
+                {
+                    await file.DeleteAsync().AsTask().ConfigureAwait(false);
+                }
+                catch
+                {
+                    // Just ignore errors for now}
+                }
+            }
+        }
+
+        /// <summary>
+        /// Initialises with default values if user has not initialised explicitly
+        /// </summary>
+        /// <returns>awaitable task</returns>
+        private async Task ForceInitialiseAsync()
+        {
+            if (_cacheFolder != null)
+            {
+                return;
+            }
+
+            await _cacheFolderSemaphore.WaitAsync().ConfigureAwait(false);
+
+            _inMemoryFileStorage = new InMemoryStorage<T>();
+
+            if (_baseFolder == null)
+            {
+                _baseFolder = ApplicationData.Current.TemporaryFolder;
+            }
+
+            if (string.IsNullOrWhiteSpace(_cacheFolderName))
+            {
+                _cacheFolderName = GetType().Name;
+            }
+
+            try
+            {
+                _cacheFolder = await ApplicationData.Current.TemporaryFolder.CreateFolderAsync(_cacheFolderName, CreationCollisionOption.OpenIfExists).AsTask().ConfigureAwait(false);
```

Comment:
when you instantiate the CacheBase you get a fully fledged Cache instance and that is no _baseFolder in that instance :)